lib/check_board.rb

```diff
@@ -1,27 +1,116 @@
 # frozen_string_literal: true
 
 class CheckBoard
-  def initialize
+  def initialize(player_gateway: player_gateway)
+    @player_gateway = player_gateway
     @score = ''
+    @check_score = @player_gateway.get_board
+    @board = [['-', '-', '-'], ['-', '-', '-'], ['-', '-', '-']]
   end
 
-  def execute(*board)
-    @check_score = board.reduce
-    if @check_score.nil?
-      @score = ''
+  def execute(*)
+    if @player_gateway.get_board.nil?
```

Comment:
could you use a guard clause here or the ternary operator instead of the if statement?